### PR TITLE
Add --jobs=x to trial runs in Tox

### PR DESCRIPTION
**Pull request by wsanchez**

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -73,10 +73,10 @@ setenv =
 
 commands =
     # Run trial without coverage
-    test: trial --random=0 --logfile="{envlogdir}/trial.log" --temp-directory="{envlogdir}/trial.d" {posargs:{env:PY_MODULE}}
+    test: trial --random=0 --jobs=8 --logfile="{envlogdir}/trial.log" --temp-directory="{envlogdir}/trial.d" {posargs:{env:PY_MODULE}}
```
Comment (adiroiban) [inline tox.ini:76]: why hardcode to 8 jobs ? ...  GitHub CI only have 2 CPUs 

I guess that we need `--jobs=auto` in trial :)

Comment (wsanchez) [reply tox.ini:76]: auto would be best, yes. Will switch to 2

**Comment by wsanchez** [comment]
Review is welcome, but I'm going to leave this PR open for a while to see if we get any new transient errors as a result of it.

**Comment by wsanchez** [comment]
Gathering some data points on runs in this PR compares to the most recent merge to master…

 * py39-twcurrent: `Ran 242 tests in 44.833s` vs. `Ran 242 tests in 196.413s`
 * pypy3-twcurrent: `Ran 242 tests in 20.278s` vs. `Ran 242 tests in 647.919s`

File: tox.ini (hunk 1 above)
Comment (adiroiban) [reply tox.ini]: See https://github.com/twisted/twisted/pull/1545#discussion_r591966698

Glyph raised a valid point. If the test are heave on IO you might want to run with `-j 4` :)
On Twisted,  I got a 10% improvement with -j4 over -j2 ... but with -j4 pypy tests were always failing

So -j8 might not be such a bad value for GitHub Actions :) and the speed improvement is amazing

**Comment by altendky** [comment]
Any tests not using [`twisted.internet.task.Clock`](https://twistedmatrix.com/documents/21.2.0/api/twisted.internet.task.Clock.html) but still using sleep and timeouts and such would benefit from going beyond two jobs.

**Comment by wsanchez** [comment]
It seems like the difference between 1 and 2 is rather profound, and there difference between 2 and N will be significantly less so… I'm inclined to leave this at 2 for this PR and we can open another PR to measure a larger number later, but also meh.

**Comment by wsanchez** [comment]
OK this is consistently better by an order of magnitude.

Can I get a green light?

**Review by adiroiban — APPROVED** [review]
Ready to land :) Thanks!

**Pull request merged**